app/components/SignUp/BillingInfo/BillingInfoForm.js

```diff
@@ -44,14 +45,38 @@ export class BillingInfoForm extends React.Component {
     });
   };
 
-  onSubmit = () => {
+  submitAddressValidation = async (values) => {
+    let validationErrors = {
+      billingInfo: {
+        address: {
+        }
+      }
+    };
+    await this.props.checkAddress(values.billingInfo.address);
+
+    if (!this.props.addressValidation.valid) {
+      this.props.addressValidation.errorMsg.forEach((error) => {
+        const fieldName = (error.name).toLowerCase();
+        validationErrors = {
+          ...validationErrors,
+          billingInfo: {
+            ...validationErrors.billingInfo,
+            address: {
+              ...validationErrors.billingInfo.address,
+              [fieldName]: error.description
+            }
+          }
+        };
+      });
+      throw new SubmissionError(validationErrors);
+    }
```

Comment:
This might be a simpler way to do the same thing.
```suggestion
    await this.props.checkAddress(values.billingInfo.address);
    const validationErrors = { 
      billingInfo: { 
        address: {}
      }
    };
    if (!this.props.addressValidation.valid) {
      this.props.addressValidation.errorMsg.forEach((error) => {
        validationErrors.billingInfo.address[error.name.toLowerCase()] = error.description;
      });
      throw new SubmissionError(validationErrors);
    }
```